### ISSUE TITLE
0.9.1 broken on Windows?

### ISSUE THREAD

**Issue by dhirschfeld** [jupyterlab/jupyterlab#1265]
After conda installing the latest 0.9.1 and running `jupyter lab` the app starts in my browser but I can't open any notebooks or code editors. The console does appear to work. Neither clicking on the launcher icons nor using the menu items seems to work to open either a notebook or text editor

```
C:\dev\code\sandbox>jupyter lab
[I 10:33:17.794 LabApp] [nb_conda_kernels] enabled, 2 kernels found
[I 10:33:18.154 LabApp] \u2713 nbpresent HTML export ENABLED
[W 10:33:18.154 LabApp] \u2717 nbpresent PDF export DISABLED: No module named 'nbbrowserpdf'
[I 10:33:18.359 LabApp] [nb_anacondacloud] enabled
[I 10:33:18.364 LabApp] [nb_conda] enabled
[I 10:33:18.364 LabApp] JupyterLab alpha preview extension loaded from C:\bin\Anaconda\lib\site-packages\jupyterlab
[I 10:33:18.455 LabApp] Serving notebooks from local directory: C:\dev\code\sandbox
[I 10:33:18.455 LabApp] 0 active kernels
[I 10:33:18.456 LabApp] The Jupyter Notebook is running at: http://localhost:8888/
[I 10:33:18.456 LabApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[W 10:33:19.368 LabApp] 404 GET /api/terminals?1479601999321 (::1) 32.40ms referer=http://localhost:8888/lab
[W 10:33:19.373 LabApp] 404 GET /api/terminals?1479601999360 (::1) 3.50ms referer=http://localhost:8888/lab
[W 10:33:19.592 LabApp] Session not found: session_id='2a3f89f5-3377-478e-bdb0-4901d61fd62f'
[W 10:33:19.595 LabApp] 404 GET /api/sessions/2a3f89f5-3377-478e-bdb0-4901d61fd62f?1479601999587 (::1) 3.80ms referer=http://localhost:8888/lab
[W 10:33:19.598 LabApp] Session not found: session_id='87ea46e2-cd9d-4990-8eb0-5e3cc5c0e038'
[W 10:33:19.598 LabApp] 404 GET /api/sessions/87ea46e2-cd9d-4990-8eb0-5e3cc5c0e038?1479601999587 (::1) 1.50ms referer=http://localhost:8888/lab
[W 10:33:29.331 LabApp] 404 GET /api/terminals?1479602009322 (::1) 2.98ms referer=http://localhost:8888/lab
[W 10:33:39.341 LabApp] 404 GET /api/terminals?1479602019323 (::1) 6.50ms referer=http://localhost:8888/lab
[I 10:33:46.249 LabApp] Kernel started: 80d0e6f1-1339-47e8-9771-77910cebd465
[W 10:33:49.339 LabApp] 404 GET /api/terminals?1479602029323 (::1) 10.02ms referer=http://localhost:8888/lab
[W 10:33:59.361 LabApp] 404 GET /api/terminals?1479602039323 (::1) 7.00ms referer=http://localhost:8888/lab
[W 10:34:09.352 LabApp] 404 GET /api/terminals?1479602049324 (::1) 14.00ms referer=http://localhost:8888/lab
[W 10:34:19.332 LabApp] 404 GET /api/terminals?1479602059324 (::1) 4.00ms referer=http://localhost:8888/lab
[W 10:34:29.354 LabApp] 404 GET /api/terminals?1479602069324 (::1) 11.51ms referer=http://localhost:8888/lab
[W 10:34:39.352 LabApp] 404 GET /api/terminals?1479602079326 (::1) 14.00ms referer=http://localhost:8888/lab
[W 10:34:49.330 LabApp] 404 GET /api/terminals?1479602089324 (::1) 2.97ms referer=http://localhost:8888/lab
[W 10:34:59.353 LabApp] 404 GET /api/terminals?1479602099324 (::1) 12.05ms referer=http://localhost:8888/lab
[W 10:35:09.409 LabApp] 404 GET /api/terminals?1479602109323 (::1) 3.00ms referer=http://localhost:8888/lab
[W 10:35:19.332 LabApp] 404 GET /api/terminals?1479602119323 (::1) 4.03ms referer=http://localhost:8888/lab
[W 10:35:29.373 LabApp] 404 GET /api/terminals?1479602129323 (::1) 11.05ms referer=http://localhost:8888/lab
[W 10:35:39.374 LabApp] 404 GET /api/terminals?1479602139323 (::1) 10.54ms referer=http://localhost:8888/lab
[W 10:35:49.355 LabApp] 404 GET /api/terminals?1479602149322 (::1) 7.00ms referer=http://localhost:8888/lab
[W 10:35:59.374 LabApp] 404 GET /api/terminals?1479602159323 (::1) 10.57ms referer=http://localhost:8888/lab
[W 10:36:09.348 LabApp] 404 GET /api/terminals?1479602169323 (::1) 12.00ms referer=http://localhost:8888/lab
[I 10:36:13.464 LabApp] Interrupted...
[I 10:36:13.465 LabApp] Shutting down kernels
[I 10:36:14.071 LabApp] Kernel shutdown: 80d0e6f1-1339-47e8-9771-77910cebd465
```

**Comment by blink1073** [comment]
Hi @dhirschfeld, I'm seeing the same thing, debugging now.

**Comment by blink1073** [comment]
The problem is that we are not respecting the `supports_terminals` metadata from the notebook server.  We are waiting for the service manager to signal that it is ready before creating new notebooks and text files, and it is never getting into that state because the terminals being ready is part of that criteria.  This will require a fix in `@jupyterlab/services` to check for the `supports_terminals` metadata, and an update to JupyterLab to get that metadata from the base notebook app and make it available to services.  We should also check for that metadata before allowing the terminal plugin to activate itself and using terminal apis in the wider application.

**Comment by dhirschfeld** [comment]
@blink1073 thanks for looking into it and sorry I didn't get around to checking *before* the release :|

Regarding the terminals, I'm running WSL and have gotten terminado up and running previously. Would it be possible to enable the jupyterlab terminals on top of WSL do you think? If so, do you have any pointers where I should look to see what is involved? I've had a quick search but am not really sure where to begin...

**Comment by blink1073** [comment]
The notebook server leaves it up to terminado whether terminals are available or not, so fixing it in terminado should do the trick:  https://github.com/jupyter/notebook/blob/e6fb7e65e94f6dd94fc5aba0cbe4bd3065e19623/notebook/notebookapp.py#L1107-L1110

**Comment by stonebig** [comment]
terminado is now ok for Windows ?

**Comment by blink1073** [comment]
No, AFAIK terminado has not yet gotten WSL support.

**Comment by blink1073** [comment]
I released 0.10 and confirmed that it works from a fresh conda environment on Windows 10.
